app/src/test/java/chat/rocket/android/authentication/loginoptions/presentation/LoginOptionsPresenterTest.kt

```diff
@@ -0,0 +1,96 @@
+package chat.rocket.android.authentication.loginoptions.presentation
+
+import chat.rocket.android.analytics.AnalyticsManager
+import chat.rocket.android.authentication.presentation.AuthenticationNavigator
+import chat.rocket.android.core.lifecycle.CancelStrategy
+import chat.rocket.android.infrastructure.LocalRepository
+import chat.rocket.android.server.domain.*
+import chat.rocket.android.server.domain.model.Account
+import chat.rocket.android.server.infrastructure.RocketChatClientFactory
+import chat.rocket.core.RocketChatClient
+import org.junit.Before
+import org.junit.Test
+import org.mockito.Mockito
+import org.mockito.Mockito.verify
+import org.mockito.MockitoAnnotations
+
+
+class LoginOptionsPresenterTest {
+
+    private var currentServer = "https://open.rocket.chat"
+    private val serverUrl = "serverUrl"
+    private val userName = "userName"
+    private val avatar = "serverUrl/avatar/userName?format=jpeg&rc_uid=null&rc_token=null"
+
+    private val account = Account(serverUrl, serverUrl, null,
+        null, userName, avatar)
+
+    private val view = Mockito.mock(LoginOptionsView::class.java)
+    private val strategy = Mockito.mock(CancelStrategy::class.java)
+    private val navigator = Mockito.mock(AuthenticationNavigator::class.java)
+    private val localRepository = Mockito.mock(LocalRepository::class.java)
+    private val settingsInteractor = Mockito.mock(GetSettingsInteractor::class.java)
+    private val analyticsManager = Mockito.mock(AnalyticsManager::class.java)
+    private val saveCurrentServerInteractor = Mockito.mock(SaveCurrentServerInteractor::class.java)
+    private val saveAccountInteractor = Mockito.mock(SaveAccountInteractor::class.java)
+    private val factory = Mockito.mock(RocketChatClientFactory::class.java)
+    private val serverInteractor = Mockito.mock(GetConnectingServerInteractor::class.java)
+    private val tokenRepository = Mockito.mock(TokenRepository::class.java)
+    private lateinit var settings: PublicSettings
+    private val token = tokenRepository.get(currentServer)
+    private lateinit var client: RocketChatClient
+
+    lateinit var loginOptionsPresenter: LoginOptionsPresenter
+
+    @Before
+    fun setUp() {
+        MockitoAnnotations.initMocks(this)
+        loginOptionsPresenter = LoginOptionsPresenter(view, strategy, factory, navigator, settingsInteractor, localRepository,
+            saveCurrentServerInteractor, saveAccountInteractor, analyticsManager, tokenRepository, serverInteractor)
+    }
+
+    @Test
+    fun check_create_account() {
+        loginOptionsPresenter.toCreateAccount()
+        verify(navigator).toCreateAccount()
+    }
+
+    @Test
+    fun check_login_with_email() {
+        loginOptionsPresenter.toLoginWithEmail()
+        verify(navigator).toLogin(currentServer)
+    }
+
+    @Test
+    fun check_authentication_with_oAuth() {
```

Comment:
I like that you have put placeholders here for tests that have not been written yet.  But should you comment them out until they have been written by somebody?  If not, do they look like they have passed, or are they just ignored?

Would it be useful to do this in all the files you are working in?  That way it will be very clear which tests still need to be written, and it might serve as a guide moving forward?

Reply:
Yes, currently they will be passing by default. I will update them with the tests very soon. 

> Would it be useful to do this in all the files you are working in? That way it will be very clear which tests still need to be written, and it might serve as a guide moving forward?

Ok, I will write if some test cases are missing by now and can be written in future

Reply:
Thanks.... Maybe I am not understanding this correctly.....  but, you might want to comment out ```@Test``` temporarily until they have been written, so that it does not look like our coverage is inflated.

Reply:
Sure, I will comment them in next commit or write the tests for it, Though empty tests will not contribute toward coverage, the coverage is generated by the code running inside the tests. Since they are empty they would only be increasing the test count.

Reply:
I see....  makes sense.......  thanks.

Reply:
@ear-dev These tests need to be removed as they were using doAuthentication method which is a coroutine method.